src/KKTSolver/krylov.jl

```diff
@@ -0,0 +1,239 @@
+        )
+    end
+end
+
+setup(::Type{KrylovPDSolver}, A; method, kwargs...) = KrylovPDSolver(method, A; kwargs...)
+
+backend(kkt::KrylovPDSolver) = "Krylov ($(kkt.f))"
+linear_system(::KrylovPDSolver) = "Normal equations"
+
+
+"""
+    update!(kkt::KrylovPDSolver, θ, regP, regD)
+
+Update diagonal scaling ``θ`` and primal-dual regularizations.
+"""
+function update!(
+    kkt::KrylovPDSolver{Tv},
+    θ::AbstractVector{Tv},
+    regP::AbstractVector{Tv},
+    regD::AbstractVector{Tv}
+) where{Tv<:Real}
+    # Sanity checks
+    length(θ)  == kkt.n || throw(DimensionMismatch(
+        "θ has length $(length(θ)) but linear solver is for n=$(kkt.n)."
+    ))
+    length(regP) == kkt.n || throw(DimensionMismatch(
+        "regP has length $(length(regP)) but linear solver has n=$(kkt.n)"
+    ))
+    length(regD) == kkt.m || throw(DimensionMismatch(
+        "regD has length $(length(regD)) but linear solver has m=$(kkt.m)"
+    ))
+
+    # Update diagonal scaling
+    kkt.θ .= θ
+    # Update regularizers
+    kkt.regP .= regP
+    kkt.regD .= regD
+
+    return nothing
+end
+
+"""
+    solve!(dx, dy, kkt::KrylovPDSolver, ξp, ξd)
+
+Solve the normal equations system using an iterative method.
+"""
+function solve!(
+    dx::Vector{Tv}, dy::Vector{Tv},
+    kkt::KrylovPDSolver{Tv},
+    ξp::Vector{Tv}, ξd::Vector{Tv}
+) where{Tv<:Real}
+    m, n = kkt.m, kkt.n
+
+    # Setup
+    d = one(Tv) ./ (kkt.θ .+ kkt.regP)
+    D = Diagonal(d)
+    
+    # Set-up right-hand side
+    ξ_ = ξp .+ kkt.A * (D * ξd)
+
+    # Form linear operator S = A * D * A' + Rd
+    # Here we pre-allocate the intermediary and final vectors
+    v1 = zeros(Tv, n)
+    v2 = zeros(Tv, m)
+    opS = LO.LinearOperator(Tv, m, m, true, true,
+        w -> begin
+            mul!(v1, kkt.A', w)
+            v1 .*= d
+            mul!(v2, kkt.A, v1)
+            v2 .+= kkt.regD .* w
+            v2
+        end
+    )
+
+    # Solve normal equations
+    _dy, stats = kkt.f(opS, ξ_, atol=kkt.atol, rtol=kkt.rtol)
+    dy .= _dy
+
+    # Recover dx
+    dx .= D * (kkt.A' * dy - ξd)
+
+    return nothing
+end
+
+
+# ==============================================================================
+#   KrylovSQDSolver: 
+# ==============================================================================
+
+"""
+    KrylovSQDSolver{T, F, Tv, Ta}
+
+Solves the augmented system with an iterative method `f::F`.
+
+```julia
+model.params.KKTOptions = Tulip.KKT.SolverOptions(
+    KrylovSQDSolver, method=Krylov.trimr
+)
+```
+"""
+mutable struct KrylovSQDSolver{T, F, Tv, Ta} <: KrylovSolver{T}
+    m::Int
+    n::Int
+    
+    f::F     # Krylov function
+    atol::T  # absolute tolerance
+    rtol::T  # relative tolerance
+
+    # Problem data
+    A::Ta     # Constraint matrix
+    θ::Tv     # scaling
+    regP::Tv  # primal regularization
+    regD::Tv  # dual regularization
+
+    function KrylovSQDSolver(f::Function, A::Ta;
+        atol::T=sqrt(eps(T)),
+        rtol::T=sqrt(eps(T))
```

Comment:
We should use `eps(T)^(2/3)` here.

Reply:
Why the choice of `2/3`?

Reply:
Because `1/2` is too small ^^

Reply:
Because `1/2` is not a sufficient tolerance for interior-point methods. `2/3` or `3/4` are more relevant for this application.